File: Python_services/syllable_Comparision/akshara_splitting.py
def is_kannada_consonant(ch):
    return '\u0C95' <= ch <= '\u0CB9'   # ಕ to ಹ

def is_kannada_vowel_sign(ch):
    return '\u0CBE' <= ch <= '\u0CCC' or ch in ['\u0CD5', '\u0CD6']  # ಾ to ೌ + ೕ ೖ

def is_kannada_diacritic(ch):
    return ch in ['\u0C82', '\u0C83']  # ಂ (anusvara), ಃ (visarga)

def split_into_aksharas(word):
    aksharas = []
    i = 0

    while i < len(word):
        akshara = word[i]     # First letter
        i += 1

        # Handle consonant clusters: ತ್ + ತ , ಭ್ + ಯ etc.
        while (
            i + 1 < len(word) and
            word[i] == '್' and
            is_kannada_consonant(word[i + 1])
        ):
            akshara += word[i]      # halant
            akshara += word[i + 1]  # next consonant
            i += 2

        # Add vowel sign (ಾ ಿ ೀ ು ೂ ೆ ೇ ೈ ೊ ೋ ೌ)
        if i < len(word) and is_kannada_vowel_sign(word[i]):
            akshara += word[i]
            i += 1

        # Add diacritics like ಂ or ಃ
        while i < len(word) and is_kannada_diacritic(word[i]):
            akshara += word[i]
            i += 1

        aksharas.append(akshara)

    return aksharas

File: Python_services/syllable_Comparision/akshara_splitting.js
function splitIntoAksharas(word) {
  const aksharas = [];
  let i = 0;

  while (i < word.length) {
    let akshara = '';
    akshara += word[i]; // First letter
    i++;

    // Build consonant clusters like ಭ್ + ಯ
    while (i + 1 < word.length && word[i] === '್' && isKannadaConsonant(word[i + 1])) {
      akshara += word[i];     // Add halant
      akshara += word[i + 1]; // Add next consonant
      i += 2;
    }

    // Add vowel sign if present
    if (i < word.length && isKannadaVowelSign(word[i])) {
      akshara += word[i];
      i++;
    }

    aksharas.push(akshara);
  }

  return aksharas;
}

function isKannadaConsonant(ch) {
  return ch >= '\u0C95' && ch <= '\u0CB9'; // ಕ to ಹ
}

function isKannadaVowelSign(ch) {
  return /[\u0CBE-\u0CCC\u0CD5\u0CD6]/.test(ch); // ಾ to ೌ, etc.
}

function launchConfetti() {
  const confetti = document.createElement('div');
  confetti.className = 'confetti';
  document.body.appendChild(confetti);
  setTimeout(() => confetti.remove(), 2000);
}

